.github/workflows/docker-release.yml

```diff
@@ -0,0 +1,90 @@
+        steps:
+            - name: Checkout repository
+              id: checkout
+              uses: actions/checkout@v3
+
+            - name: Install Docker BuildX
+              uses: docker/setup-buildx-action@v2.7.0
+              id: buildx
+              with:
+                  install: true
+
+            # Login against a Docker registry except on PR
+            # https://github.com/docker/login-action
+            - name: Log into registry ${{ env.REGISTRY }}
+              # Ensure this doesn't trigger on PR's
+              if: github.event_name != 'pull_request'
+              uses: docker/login-action@v2.2.0
+              with:
+                  registry: ${{ env.REGISTRY }}
+                  username: ${{ github.actor }}
+                  password: ${{ secrets.GITHUB_TOKEN }}
+
+            # Extract metadata (tags, labels) for Docker
+            # https://github.com/docker/metadata-action
+            - name: Extract Docker metadata
+              id: meta
+              uses: docker/metadata-action@v4.6.0
+              with:
+                  images: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}
+
+            # Creates an additional 'latest' or 'nightly' tag
+            # If the job is triggered via cron schedule, tag nightly and nightly-{SHA}
+            # If the job is triggered via workflow dispatch and on a master branch, tag branch and latest
+            # Otherwise, just tag as the branch name
+            - name: Finalize Docker Metadata
+              id: docker_tagging
+              run: |
+                    echo "Tagging as branch name"
+                    echo "::set-output name=docker_tags::${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}:${GITHUB_REF##*/}"
+
+            # Log docker metadata to explicitly know what is being pushed
+            - name: Inspect Docker Metadata
+              run: |
+                  echo "TAGS -> ${{ steps.docker_tagging.outputs.docker_tags }}"
+                  echo "LABELS ->  ${{ steps.meta.outputs.labels }}"
+
+            # Build and push Docker image
+            # https://github.com/docker/build-push-action
+            # https://github.com/docker/build-push-action/blob/master/docs/advanced/cache.md
+            - name: Build and push Docker image
+              uses: docker/build-push-action@v3
+              with:
+                  context: .
+                  push: true
+                  tags: ${{ steps.docker_tagging.outputs.docker_tags }}
+                  labels: ${{ steps.meta.outputs.labels }}
+                  cache-from: type=gha
+                  cache-to: type=gha,mode=max
+                  build-args: |
+                      BUILDTIME=${{ fromJSON(steps.meta.outputs.json).labels['org.opencontainers.image.created'] }}
+                      VERSION=${{ fromJSON(steps.meta.outputs.json).labels['org.opencontainers.image.version'] }}
+                      REVISION=${{ fromJSON(steps.meta.outputs.json).labels['org.opencontainers.image.revision'] }}
```

Comment:
What is the purpose of pushing this as a docker image?  This project builds a browser extension, which doesn't really make sense to use inside of a docker image that I can think of.  We already do builds inside of docker for reproducibility/consistency, but the image is discarded upon build completion.